# SATVIK SHARMA
# B.Tech 1st Year

---
Full ML Project (Frontend, Backend, Regression Models)


In [1]:
# Importing Libraries and Dataset
import pandas as pd
import numpy as np

df = pd.read_csv("Original_laptop_data.csv", encoding='latin1')

# Understanding and analysing the Dataset:

In [2]:
df.head(10)

,laptop_ID,Company,Product,TypeName,Inches,ScreenResolution,Cpu,Ram,Memory,Gpu,OpSys,Weight,Price_euros
0,1,Apple,MacBook Pro,Ultrabook,13.3,IPS Panel Retina Display 2560x1600,Intel Core i5 2.3GHz,8GB,128GB SSD,Intel Iris Plus Graphics 640,macOS,1.37kg,1339.69
1,2,Apple,Macbook Air,Ultrabook,13.3,1440x900,Intel Core i5 1.8GHz,8GB,128GB Flash Storage,Intel HD Graphics 6000,macOS,1.34kg,898.94
2,3,HP,250 G6,Notebook,15.6,Full HD 1920x1080,Intel Core i5 7200U 2.5GHz,8GB,256GB SSD,Intel HD Graphics 620,No OS,1.86kg,575.00
3,4,Apple,MacBook Pro,Ultrabook,15.4,IPS Panel Retina Display 2880x1800,Intel Core i7 2.7GHz,16GB,512GB SSD,AMD Radeon Pro 455,macOS,1.83kg,2537.45
4,5,Apple,MacBook Pro,Ultrabook,13.3,IPS Panel Retina Display 2560x1600,Intel Core i5 3.1GHz,8GB,256GB SSD,Intel Iris Plus Graphics 650,macOS,1.37kg,1803.60
5,6,Acer,Aspire 3,Notebook,15.6,1366x768,AMD A9-Series 9420 3GHz,4GB,500GB HDD,AMD Radeon R5,Windows 10,2.1kg,400.00
6,7,Apple,MacBook Pro,Ultrabook,15.4,IPS Panel Retina Display 2880x1800,Intel Core i7 2.2GHz,16GB,256GB Flash Storage,Intel Iris Pro Graphics,Mac OS X,2.04kg,2139.97
7,8,Apple,Macbook Air,Ultrabook,13.3,1440x900,Intel Core i5 1.8GHz,8GB,256GB Flash Storage,Intel HD Graphics 6000,macOS,1.34kg,1158.70
8,9,Asus,ZenBook UX430UN,Ultrabook,14.0,Full HD 1920x1080,Intel Core i7 8550U 1.8GHz,16GB,512GB SSD,Nvidia GeForce MX150,Windows 10,1.3kg,1495.00
9,10,Acer,Swift 3,Ultrabook,14.0,IPS Panel Full HD 1920x1080,Intel Core i5 8250U 1.6GHz,8GB,256GB SSD,Intel UHD Graphics 620,Windows 10,1.6kg,770.00


In [3]:
df.describe()

,laptop_ID,Inches,Price_euros
count,1303.000000,1303.000000,1303.000000
mean,660.155794,15.017191,1123.686992
std,381.172104,1.426304,699.009043
min,1.000000,10.100000,174.000000
25%,331.500000,14.000000,599.000000
50%,659.000000,15.600000,977.000000
75%,990.500000,15.600000,1487.880000
max,1320.000000,18.400000,6099.000000


In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1303 entries, 0 to 1302
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   laptop_ID         1303 non-null   int64  
 1   Company           1303 non-null   str    
 2   Product           1303 non-null   str    
 3   TypeName          1303 non-null   str    
 4   Inches            1303 non-null   float64
 5   ScreenResolution  1303 non-null   str    
 6   Cpu               1303 non-null   str    
 7   Ram               1303 non-null   str    
 8   Memory            1303 non-null   str    
 9   Gpu               1303 non-null   str    
 10  OpSys             1303 non-null   str    
 11  Weight            1303 non-null   str    
 12  Price_euros       1303 non-null   float64
dtypes: float64(2), int64(1), str(10)
memory usage: 132.5 KB


In [5]:
df.shape

(1303, 13)

In [6]:
df.duplicated().sum()

np.int64(0)

## Observations:
1. There are 4 Datatypes: float64(2), int64(1), object(10).
2. Memory Usage is 132.5+ KB.
3. There are a total of 1303 Rows & 13 Columns.
4. There are no Duplicate or Empty Columns/Rows.

# Preprocessing Begins:

In [7]:
# Removing the column 'laptop_ID', as it is just serial number.
df.drop(columns = ['laptop_ID'], inplace = True)

In [8]:
# Removing 'GB' from the whole Ram Column and Converting it to Integer for better memory allocation and clean data.
df['Ram'] = df['Ram'].str.replace('GB', '', regex=False)
df['Ram'] = df['Ram'].astype(int)

In [9]:
# Removing 'kg' from the whole Weight Column and Converting it to Integer for better memory allocation and clean data.
df['Weight'] = df['Weight'].str.replace('kg', '', regex=False)
df['Weight'] = df['Weight'].astype(float)

In [10]:
# Trying to Standardise the 'ScreenResolution' Column
df["ScreenResolution"].value_counts()

ScreenResolution
Full HD 1920x1080                                507
1366x768                                         281
IPS Panel Full HD 1920x1080                      230
IPS Panel Full HD / Touchscreen 1920x1080         53
Full HD / Touchscreen 1920x1080                   47
1600x900                                          23
Touchscreen 1366x768                              16
Quad HD+ / Touchscreen 3200x1800                  15
IPS Panel 4K Ultra HD 3840x2160                   12
IPS Panel 4K Ultra HD / Touchscreen 3840x2160     11
4K Ultra HD / Touchscreen 3840x2160               10
IPS Panel 1366x768                                 7
Touchscreen 2560x1440                              7
4K Ultra HD 3840x2160                              7
IPS Panel Retina Display 2560x1600                 6
IPS Panel Retina Display 2304x1440                 6
Touchscreen 2256x1504                              6
IPS Panel Quad HD+ / Touchscreen 3200x1800         6
IPS Panel Touchscreen 2560x14

In [11]:
df['resolution'] = df['ScreenResolution'].str.split().str[-1]

df['X_res'] = df['resolution'].str.split('x').str[0].astype(int)
df['Y_res'] = df['resolution'].str.split('x').str[1].astype(int)

In [12]:
df['Touchscreen'] = df['ScreenResolution'].apply(
    lambda x: 1 if 'Touchscreen' in x else 0
)

In [13]:
df['IPS'] = df['ScreenResolution'].apply(
    lambda x: 1 if 'IPS' in x else 0
)

In [14]:
def get_screen_type(x):
    if '4K' in x or 'Ultra HD' in x:
        return '4K'
    elif 'Full HD' in x:
        return 'Full HD'
    elif 'HD' in x:
        return 'HD'
    else:
        return 'Other'

df['Screen_Type'] = df['ScreenResolution'].apply(get_screen_type)

In [15]:
df.drop(columns=['ScreenResolution', 'resolution'], inplace=True)

In [16]:
# Using Numpy, Creating a pixel density column for better price prediction
df['ppi'] = (((df['X_res']**2 + df['Y_res']**2)**0.5) / df['Inches'])

In [17]:
df.drop(columns=['X_res','Y_res','Inches'], inplace=True)

In [18]:
df.head(1)

,Company,Product,TypeName,Cpu,Ram,Memory,Gpu,OpSys,Weight,Price_euros,Touchscreen,IPS,Screen_Type,ppi
0,Apple,MacBook Pro,Ultrabook,Intel Core i5 2.3GHz,8,128GB SSD,Intel Iris Plus Graphics 640,macOS,1.37,1339.69,0,1,Other,226.983005


In [19]:
df.head()

,Company,Product,TypeName,Cpu,Ram,Memory,Gpu,OpSys,Weight,Price_euros,Touchscreen,IPS,Screen_Type,ppi
0,Apple,MacBook Pro,Ultrabook,Intel Core i5 2.3GHz,8,128GB SSD,Intel Iris Plus Graphics 640,macOS,1.37,1339.69,0,1,Other,226.983005
1,Apple,Macbook Air,Ultrabook,Intel Core i5 1.8GHz,8,128GB Flash Storage,Intel HD Graphics 6000,macOS,1.34,898.94,0,0,Other,127.677940
2,HP,250 G6,Notebook,Intel Core i5 7200U 2.5GHz,8,256GB SSD,Intel HD Graphics 620,No OS,1.86,575.00,0,0,Full HD,141.211998
3,Apple,MacBook Pro,Ultrabook,Intel Core i7 2.7GHz,16,512GB SSD,AMD Radeon Pro 455,macOS,1.83,2537.45,0,1,Other,220.534624
4,Apple,MacBook Pro,Ultrabook,Intel Core i5 3.1GHz,8,256GB SSD,Intel Iris Plus Graphics 650,macOS,1.37,1803.60,0,1,Other,226.983005


In [20]:
df["Cpu"].value_counts()

Cpu
Intel Core i5 7200U 2.5GHz              190
Intel Core i7 7700HQ 2.8GHz             146
Intel Core i7 7500U 2.7GHz              134
Intel Core i7 8550U 1.8GHz               73
Intel Core i5 8250U 1.6GHz               72
                                       ... 
Intel Core i5 7200U 2.7GHz                1
Intel Pentium Dual Core 4405Y 1.5GHz      1
Intel Core M 6Y54 1.1GHz                  1
Samsung Cortex A72&A53 2.0GHz             1
AMD E-Series 9000 2.2GHz                  1
Name: count, Length: 118, dtype: int64

In [21]:
df["Gpu"].value_counts()

Gpu
Intel HD Graphics 620      281
Intel HD Graphics 520      185
Intel UHD Graphics 620      68
Nvidia GeForce GTX 1050     66
Nvidia GeForce GTX 1060     48
                          ... 
Nvidia Quadro M500M          1
AMD Radeon R7 M360           1
Nvidia Quadro M3000M         1
Nvidia GeForce 960M          1
ARM Mali T860 MP4            1
Name: count, Length: 110, dtype: int64

In [22]:
df["Memory"].value_counts()

Memory
256GB SSD                        412
1TB HDD                          223
500GB HDD                        132
512GB SSD                        118
128GB SSD +  1TB HDD              94
128GB SSD                         76
256GB SSD +  1TB HDD              73
32GB Flash Storage                38
2TB HDD                           16
64GB Flash Storage                15
512GB SSD +  1TB HDD              14
1TB SSD                           14
256GB SSD +  2TB HDD              10
1.0TB Hybrid                       9
256GB Flash Storage                8
16GB Flash Storage                 7
32GB SSD                           6
180GB SSD                          5
128GB Flash Storage                4
16GB SSD                           3
512GB SSD +  2TB HDD               3
256GB SSD +  256GB SSD             2
256GB SSD +  500GB HDD             2
128GB SSD +  2TB HDD               2
1TB SSD +  1TB HDD                 2
512GB Flash Storage                2
64GB SSD                       

In [23]:
df["Product"].value_counts()

Product
XPS 13                                30
Inspiron 3567                         29
250 G6                                21
Legion Y520-15IKBN                    19
Vostro 3568                           19
                                      ..
X541NA-GO020T (N3350/4GB/1TB/W10)      1
Rog G752VT-GC073T                      1
B51-80 (i7-6500U/8GB/1008GB/Radeon     1
GE62 Apache                            1
Yoga 500-14IBD                         1
Name: count, Length: 618, dtype: int64

In [24]:
df["Company"].value_counts()

Company
Dell         297
Lenovo       297
HP           274
Asus         158
Acer         103
MSI           54
Toshiba       48
Apple         21
Samsung        9
Razer          7
Mediacom       7
Microsoft      6
Xiaomi         4
Vero           4
Chuwi          3
Google         3
Fujitsu        3
LG             3
Huawei         2
Name: count, dtype: int64

In [25]:
df["TypeName"].value_counts()

TypeName
Notebook              727
Gaming                205
Ultrabook             196
2 in 1 Convertible    121
Workstation            29
Netbook                25
Name: count, dtype: int64

In [26]:
# CPU Brand
def get_cpu_brand(x):
    if 'Intel Core i7' in x:
        return 'i7'
    elif 'Intel Core i5' in x:
        return 'i5'
    elif 'Intel Core i3' in x:
        return 'i3'
    elif 'Intel' in x:
        return 'Other Intel'
    elif 'AMD' in x:
        return 'AMD'
    else:
        return 'Other'
df['Cpu_brand'] = df['Cpu'].apply(get_cpu_brand)
# CPU Speed (GHz)
df['Cpu_speed'] = df['Cpu'].apply(lambda x: float(x.split()[-1].replace('GHz','')))

df.drop(columns=['Cpu'], inplace=True)
df = pd.get_dummies(df, columns=['Cpu_brand'], drop_first=True)

In [27]:
df['Gpu_brand'] = df['Gpu'].apply(lambda x: x.split()[0])
df.drop(columns=['Gpu'], inplace=True)
df = pd.get_dummies(df, columns=['Gpu_brand'], drop_first=True)

In [28]:
import re
def process_memory(mem):
    ssd = 0
    hdd = 0
    items = mem.split('+')
    for item in items:
        item = item.strip()
        # Convert TB to GB
        if 'TB' in item:
            size = float(re.findall(r'\d+\.?\d*', item)[0]) * 1024
        else:
            size = float(re.findall(r'\d+\.?\d*', item)[0])
        if 'SSD' in item:
            ssd += size
        elif 'HDD' in item:
            hdd += size
    return pd.Series([ssd, hdd])
df[['SSD','HDD']] = df['Memory'].apply(process_memory)
df.drop(columns=['Memory'], inplace=True)

In [29]:
def clean_os(x):
    if 'Windows' in x:
        return 'Windows'
    elif 'Mac' in x or 'macOS' in x:
        return 'Mac'
    elif 'Linux' in x:
        return 'Linux'
    else:
        return 'Other'
df['OpSys'] = df['OpSys'].apply(clean_os)
df = pd.get_dummies(df, columns=['OpSys'], drop_first=True)

In [30]:
df.head()

,Company,Product,TypeName,Ram,Weight,Price_euros,Touchscreen,IPS,Screen_Type,ppi,...,Cpu_brand_i5,Cpu_brand_i7,Gpu_brand_ARM,Gpu_brand_Intel,Gpu_brand_Nvidia,SSD,HDD,OpSys_Mac,OpSys_Other,OpSys_Windows
0,Apple,MacBook Pro,Ultrabook,8,1.37,1339.69,0,1,Other,226.983005,...,True,False,False,True,False,128.0,0.0,True,False,False
1,Apple,Macbook Air,Ultrabook,8,1.34,898.94,0,0,Other,127.677940,...,True,False,False,True,False,0.0,0.0,True,False,False
2,HP,250 G6,Notebook,8,1.86,575.00,0,0,Full HD,141.211998,...,True,False,False,True,False,256.0,0.0,False,True,False
3,Apple,MacBook Pro,Ultrabook,16,1.83,2537.45,0,1,Other,220.534624,...,False,True,False,False,False,512.0,0.0,True,False,False
4,Apple,MacBook Pro,Ultrabook,8,1.37,1803.60,0,1,Other,226.983005,...,True,False,False,True,False,256.0,0.0,True,False,False


In [31]:
df['SSD'] = df['SSD'].astype(int)
df['HDD'] = df['HDD'].astype(int)

In [32]:
df = pd.get_dummies(df, columns=['TypeName'], drop_first=True)

In [33]:
df.head()

,Company,Product,Ram,Weight,Price_euros,Touchscreen,IPS,Screen_Type,ppi,Cpu_speed,...,SSD,HDD,OpSys_Mac,OpSys_Other,OpSys_Windows,TypeName_Gaming,TypeName_Netbook,TypeName_Notebook,TypeName_Ultrabook,TypeName_Workstation
0,Apple,MacBook Pro,8,1.37,1339.69,0,1,Other,226.983005,2.3,...,128,0,True,False,False,False,False,False,True,False
1,Apple,Macbook Air,8,1.34,898.94,0,0,Other,127.677940,1.8,...,0,0,True,False,False,False,False,False,True,False
2,HP,250 G6,8,1.86,575.00,0,0,Full HD,141.211998,2.5,...,256,0,False,True,False,False,False,True,False,False
3,Apple,MacBook Pro,16,1.83,2537.45,0,1,Other,220.534624,2.7,...,512,0,True,False,False,False,False,False,True,False
4,Apple,MacBook Pro,8,1.37,1803.60,0,1,Other,226.983005,3.1,...,256,0,True,False,False,False,False,False,True,False


In [34]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1303 entries, 0 to 1302
Data columns (total 28 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Company                1303 non-null   str    
 1   Product                1303 non-null   str    
 2   Ram                    1303 non-null   int64  
 3   Weight                 1303 non-null   float64
 4   Price_euros            1303 non-null   float64
 5   Touchscreen            1303 non-null   int64  
 6   IPS                    1303 non-null   int64  
 7   Screen_Type            1303 non-null   str    
 8   ppi                    1303 non-null   float64
 9   Cpu_speed              1303 non-null   float64
 10  Cpu_brand_Other        1303 non-null   bool   
 11  Cpu_brand_Other Intel  1303 non-null   bool   
 12  Cpu_brand_i3           1303 non-null   bool   
 13  Cpu_brand_i5           1303 non-null   bool   
 14  Cpu_brand_i7           1303 non-null   bool   
 15  Gpu_brand_ARM  

In [35]:
df["Company"].value_counts()

Company
Dell         297
Lenovo       297
HP           274
Asus         158
Acer         103
MSI           54
Toshiba       48
Apple         21
Samsung        9
Razer          7
Mediacom       7
Microsoft      6
Xiaomi         4
Vero           4
Chuwi          3
Google         3
Fujitsu        3
LG             3
Huawei         2
Name: count, dtype: int64

In [36]:
df["Screen_Type"].value_counts()

Screen_Type
Full HD    843
Other      387
4K          43
HD          30
Name: count, dtype: int64

In [37]:
df = pd.get_dummies(df, columns=[
    'Company',
    'Screen_Type'
], drop_first=True)

In [38]:
# This Changes all Boolean values to 1/0.
bool_cols = df.select_dtypes(include='bool').columns
df[bool_cols] = df[bool_cols].astype(int)

In [39]:
df.dtypes

Product                      str
Ram                        int64
Weight                   float64
Price_euros              float64
Touchscreen                int64
IPS                        int64
ppi                      float64
Cpu_speed                float64
Cpu_brand_Other            int64
Cpu_brand_Other Intel      int64
Cpu_brand_i3               int64
Cpu_brand_i5               int64
Cpu_brand_i7               int64
Gpu_brand_ARM              int64
Gpu_brand_Intel            int64
Gpu_brand_Nvidia           int64
SSD                        int64
HDD                        int64
OpSys_Mac                  int64
OpSys_Other                int64
OpSys_Windows              int64
TypeName_Gaming            int64
TypeName_Netbook           int64
TypeName_Notebook          int64
TypeName_Ultrabook         int64
TypeName_Workstation       int64
Company_Apple              int64
Company_Asus               int64
Company_Chuwi              int64
Company_Dell               int64
Company_Fu

In [54]:
df = df.drop(columns=['Product'])

In [63]:
df.head(9)

,Ram,Weight,Price_euros,Touchscreen,IPS,ppi,Cpu_speed,Cpu_brand_Other,Cpu_brand_Other Intel,Cpu_brand_i3,...,Company_Mediacom,Company_Microsoft,Company_Razer,Company_Samsung,Company_Toshiba,Company_Vero,Company_Xiaomi,Screen_Type_Full HD,Screen_Type_HD,Screen_Type_Other
0,8,1.37,1339.69,0,1,226.983005,2.3,0,0,0,...,0,0,0,0,0,0,0,0,0,1
1,8,1.34,898.94,0,0,127.677940,1.8,0,0,0,...,0,0,0,0,0,0,0,0,0,1
2,8,1.86,575.00,0,0,141.211998,2.5,0,0,0,...,0,0,0,0,0,0,0,1,0,0
3,16,1.83,2537.45,0,1,220.534624,2.7,0,0,0,...,0,0,0,0,0,0,0,0,0,1
4,8,1.37,1803.60,0,1,226.983005,3.1,0,0,0,...,0,0,0,0,0,0,0,0,0,1
5,4,2.10,400.00,0,0,100.454670,3.0,0,0,0,...,0,0,0,0,0,0,0,0,0,1
6,16,2.04,2139.97,0,1,220.534624,2.2,0,0,0,...,0,0,0,0,0,0,0,0,0,1
7,8,1.34,1158.70,0,0,127.677940,1.8,0,0,0,...,0,0,0,0,0,0,0,0,0,1
8,16,1.30,1495.00,0,0,157.350512,1.8,0,0,0,...,0,0,0,0,0,0,0,1,0,0


In [64]:
import pickle
%pip install scikit-learn
%pip install xgboost
import sklearn
from sklearn.model_selection import train_test_split, KFold, cross_val_score, RandomizedSearchCV
from sklearn.metrics import r2_score, mean_absolute_error, mean_squared_error
from sklearn.ensemble import RandomForestRegressor, GradientBoostingRegressor
from sklearn.linear_model import Ridge
from xgboost import XGBRegressor


[notice] A new release of pip is available: 25.3 -> 26.0.1
[notice] To update, run: pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.

[notice] A new release of pip is available: 25.3 -> 26.0.1
[notice] To update, run: pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.


In [65]:
X = df.drop('Price_euros', axis=1)
y = np.log(df['Price_euros']) # Log transformss for right-skewed pricing

In [66]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.15, random_state=42)

In [67]:
# Defining Models to Test
models = {
    "Ridge Regression": Ridge(alpha=10),
    "Random Forest": RandomForestRegressor(random_state=42),
    "Gradient Boosting": GradientBoostingRegressor(random_state=42),
    "XGBoost": XGBRegressor(random_state=42)
}

In [68]:
# 5. Baseline Cross-Validation
kf = KFold(n_splits=5, shuffle=True, random_state=42)

for name, model in models.items():
    cv_scores = cross_val_score(model, X_train, y_train, cv=kf, scoring='r2')
    print(f"{name}: Average R2 = {cv_scores.mean():.4f} (+/- {cv_scores.std():.4f})")

Ridge Regression: Average R2 = 0.8151 (+/- 0.0070)
Random Forest: Average R2 = 0.8804 (+/- 0.0099)
Gradient Boosting: Average R2 = 0.8735 (+/- 0.0076)
XGBoost: Average R2 = 0.8913 (+/- 0.0114)


In [69]:
# Hyperparameter Tuning (XGBoost)
xgb_params = {
    'n_estimators': [100, 200, 300],
    'learning_rate': [0.01, 0.05, 0.1, 0.2],
    'max_depth': [3, 5, 7, 9],
    'colsample_bytree': [0.5, 0.7, 1.0],
    'subsample': [0.6, 0.8, 1.0]
}

In [70]:
xgb_random = RandomizedSearchCV(
    estimator=XGBRegressor(random_state=42),
    param_distributions=xgb_params,
    n_iter=20,
    cv=3,
    verbose=1,
    random_state=42,
    n_jobs=-1 # Uses all available CPU cores
)

xgb_random.fit(X_train, y_train)
print(f"Best Parameters: {xgb_random.best_params_}")

Fitting 3 folds for each of 20 candidates, totalling 60 fits
Best Parameters: {'subsample': 0.6, 'n_estimators': 300, 'max_depth': 5, 'learning_rate': 0.05, 'colsample_bytree': 0.7}


In [71]:
best_model = xgb_random.best_estimator_
y_pred_log = best_model.predict(X_test)

y_test_euros = np.exp(y_test)
y_pred_euros = np.exp(y_pred_log)

mae = mean_absolute_error(y_test_euros, y_pred_euros)
rmse = np.sqrt(mean_squared_error(y_test_euros, y_pred_euros))
r2 = r2_score(y_test_euros, y_pred_euros)

print(f"R-Squared ($R^2$): {r2:.4f} (Closer to 1 is better)")
print(f"Mean Absolute Error (MAE): €{mae:.2f}")
print(f"Root Mean Squared Error (RMSE): €{rmse:.2f}")

R-Squared ($R^2$): 0.8449 (Closer to 1 is better)
Mean Absolute Error (MAE): €163.53
Root Mean Squared Error (RMSE): €297.50


In [72]:
pickle.dump(best_model, open('model.pkl', 'wb'))
pickle.dump(X.columns.tolist(), open('columns.pkl', 'wb'))
print("Done! Ready for deployment.")

Done! Ready for deployment.


In [74]:
print("XGBoost:", xgboost.__version__)
print("Scikit-Learn:", sklearn.__version__)
print("Numpy:", numpy.__version__)

XGBoost: 3.2.0
Scikit-Learn: 1.8.0
Numpy: 2.4.3
